cl/scrapers/management/commands/scrape_caselaw.py

```diff
@@ -0,0 +1,339 @@
+    reporter_id = options["reporter_id"]
+    source = options["source"]
+    allowed_types = ["jurisdictions", "courts", "reporters", "cases"]
+    base_url = "https://api.case.law"
+    version = "v1"
+    params = {}
+    endpoint = None
+    page_size = 1000
+
+    # TODO test how many cases can we collect, there's api calls limit for registered user?
+    # TODO handle error_auth_required or error_limit_exceeded when want to get casebody
+
+    if not type:
+        logger.error("You didn't provide the type. Exiting.")
+        return
+
+    if type not in allowed_types:
+        logger.error("You didn't provide the type. Exiting.")
+        return
+
+    if last_updated and last_updated_filter:
+        if last_updated_filter not in ["gt", "gte", "lt", "lte"]:
+            logger.error(
+                f"{last_updated_filter} not in options gt, gte, lt, lte."
+            )
+            return
+
+    if not last_updated and last_updated_filter:
+        logger.error(
+            "You can't set last-update-filter without passing last-update argument."
+        )
+        return
+
+    if last_updated and not last_updated_filter:
+        logger.error(
+            "You can't filter by last-update without passing last-update-filter "
+            "argument."
+        )
+        return
+
+    if full_case and not body_format:
+        logger.error(
+            "You need to set body-format argument to get full case text. Exiting."
+        )
+        return
+
+    if not full_case and body_format:
+        logger.error(
+            "You can't set body-format argument without setting the full-case "
+            "argument to True. Exiting. "
+        )
+        return
+
+    if source and source not in ["Harvard", "Fastcase"]:
+        logger.error(
+            "Invalid source. Valid sources: Harvard or Fastcase. Exiting. "
+        )
+        return
+
+    if type == "jurisdictions":
+        endpoint = f"{base_url}/{version}/jurisdictions/"
+    elif type == "courts":
+        endpoint = f"{base_url}/{version}/courts/"
+    elif type == "reporters":
+        endpoint = f"{base_url}/{version}/reporters/"
+    elif type == "cases":
+        endpoint = f"{base_url}/{version}/cases/"
+        params["page_size"] = page_size
+        if ordering:
+            params["ordering"] = ordering
+        if court:
+            params["court"] = court
+        if jurisdiction:
+            params["jurisdiction"] = jurisdiction
+        if full_case:
+            params["full_case"] = "true" if full_case else "false"
+        if body_format:
+            params["body_format"] = body_format
+        if last_updated and last_updated_filter:
+            params[f"last_updated__{last_updated_filter}"] = last_updated
+        if reporter_id:
+            params["reporter"] = reporter_id
+        if source:
+            params["source"] = source
+
+    # Directory to store data based on type
+    directory = Path(settings.MEDIA_ROOT, "case.law", type)
```

Comment:
Can you explain why this directory is set to case.law but on line 241 it sets to harvard_corpus

Reply:
It's set to case.law directory when you want to pull other data like reporters, jurisdictions, courts, from case.law API. Only cases are stored in harvard_corpus.

Reply:
Gotcha.